### PR TITLE
Added FBX

### PR DESCRIPTION
**Pull request by washifu**
- 🎉 Added `FBX` which accepts the following props
  - `fbxPath`: string
  - `fbxURL`: string
  - `actionIndex`: number
  - all `Group` props
- 📖 Added `FBX` Story

For `actionIndex` try `samba` in the story. `actionIndex` of 1 is the model in T mode (no movement).

### CODE REVIEW

File: src/groups/FBX.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
+// FBX.tsx
+import * as React from "react";
+import * as THREE from "three";
+import { FBXLoader } from "three/examples/jsm/loaders/FBXLoader";
```
Comment (mayt) [inline src/groups/FBX.tsx:4]: I wished they'd expose those utils in a package. 😢

File: src/groups/FBX.tsx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+import PropTypes from "prop-types";
+import exact from "prop-types-exact";
+import { useViewContext, useAnimationFrame } from "../utils/hooks";
+import Group, { GroupProps, GroupPropTypes } from "./Group";
+import { Label } from "../primitives";
+
+const { useEffect, useMemo, useRef, memo } = React;
```
Comment (mayt) [inline src/groups/FBX.tsx]: You can combine them on top like so.

`import React, { useEffect, useMemo, useRef, memo } from "react";`

File: src/groups/FBX.tsx [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+          materialProps
+        });
+      }
+    },
+    [group, fbxPath, fbxURL, _envMap, castShadow, receiveShadow, materialProps]
+  );
+
+  // Animation Mixer Update
+  const clock = new THREE.Clock();
+  useAnimationFrame(function updateMixer() {
```
Comment (mayt) [inline src/groups/FBX.tsx]: does it make sense to use `useFrame()` here

File: src/groups/FBX.tsx (hunk 1 above)
Comment (washifu) [reply src/groups/FBX.tsx:4]: Me too

File: src/groups/FBX.tsx (hunk 3 above)
Comment (washifu) [reply src/groups/FBX.tsx]: Yeah, this needs to execute every frame for the animations to work. Otherwise the FBX models will just be still.

Comment (mayt) [reply src/groups/FBX.tsx]: yea `useFrame` executes just before every animation frame.

File: src/groups/FBX.tsx (hunk 2 above)
Comment (washifu) [reply src/groups/FBX.tsx]: So, for some reason react-three-fiber needs `import * as React from "react";`
which doesn't work with the comma and destructuring. 
I tried a few times and reverted back to this.

Comment (mayt) [reply src/groups/FBX.tsx]: Ah its probably because of typescript...

Comment (washifu) [reply src/groups/FBX.tsx]: Ah nevermind, it is a linter error. My bad, `THREE` is the one that needs to be imported with `* as THREE`

File: src/groups/FBX.tsx (hunk 3 above)
Comment (washifu) [reply src/groups/FBX.tsx]: Ah, sorry, I misread `useFrame` for `useAnimationFrame`. Yeah, `useFrame` works here.

**Review by mayt — APPROVED** [review]

File: src/utils/hooks.tsx [hunk 4]
```diff
@@ -53,3 +60,28 @@ export function useViewContext(): any {
...
+  // without triggering a re-render on their change
+  const requestRef = useRef<number>();
+
+  function animate(): void {
+    callback();
+    requestRef.current = requestAnimationFrame(animate);
+  }
+
+  useEffect(() => {
+    requestRef.current = requestAnimationFrame(animate);
```
Comment (mayt) [inline src/utils/hooks.tsx]:
```suggestion
    animate();
```

This was a bug that caused the other jitter on local run.

**Review by mayt — CHANGES_REQUESTED** [review]
Please update `src/utils/events.tsx`

```
-import { useAnimationFrame } from "./util";
+import { useAnimationFrame } from "./hooks";
```

Seems like i cant update this PR via cmd unless I have access to your fork

**Comment by washifu** [comment]
Fixed the import in `events.tsx`
Since I do not have write access, I cannot merge. You will have to merge the PR after approving.
(Would love to become a maintainer, or receive write access 😉 )

**Comment by mayt** [comment]
I'll test and merge it then

**Pull request merged**